### PR TITLE
Add button to go back to workspaces list

### LINKED ISSUE #828: Sign out tries to redirect user to last workspace, gives error
When I sign out from Terra while viewing a workspace, I go to the hamburger menu and click "Sign Out". Then I see the sign in page, and I sign in with a new identity. Terra then tries to load the last workspace that I was viewing. However my new identity often doesn't have access, so I get a "no access" message, which is confusing. 

2 ideas for how to handle this:

1: include a button to go Home on the "Workspace not found" page. Currently it says:
```
Could not display workspace
You are trying to access a workspace that either does not exist, or you do not have access to it.

To view an existing workspace, the owner of the workspace must share it with you or with a Group of which you are a member. If the workspace is protected under an Authorization Domain, you must be a member of every group within the Authorization Domain.

If you think the workspace exists but you do not have access, please contact the workspace owner.
```
I think it should include a button to "Return to Home" at the button.

2. When a user logs out and logs back in, direct them back to #workspaces. This is instead of loading the page from where they logged out. 

I'm labeling this as a bug as it is a pretty unideal UX, though it is by design.

### PR DESCRIPTION
**Pull request by MattPutnam**
Fixes #828 

Side note: is anyone else having an issue where signing out and then trying to sign in automatically selects the most recently signed-in user (without asking)?

### CODE REVIEW

**Review by breilly2 — APPROVED** [review]
👍

**Review by panentheos — APPROVED** [review]
Looks good.

File: src/pages/workspaces/workspace/WorkspaceContainer.js [hunk 1]
```diff
@@ -233,7 +233,11 @@ export const wrapWorkspace = ({ breadcrumbs, activeTab, title, topBarContent, sh
             ]),
             p({}, [
               'If you think the workspace exists but you do not have access, please contact the workspace owner.'
-            ])
+            ]),
+            buttonPrimary({
+              as: 'a',
+              href: Nav.getLink('workspaces')
+            }, ['Workspaces List'])
```
Comment (katevoss) [inline src/pages/workspaces/workspace/WorkspaceContainer.js]: What about naming the button `Return to Workspaces`? I think it makes it a clearer call-to-action.

Comment (zarsky-broad) [reply src/pages/workspaces/workspace/WorkspaceContainer.js]: Brad's recommendation was "Return to Workspace List"

**Review by zarsky-broad — APPROVED** [review]

**Review by elizzhao — APPROVED** [review]
I think we should change it to "Return to Workspace List" as suggested, but aside from that--looks good!

**Pull request merged**